File: solutions/python/robot-simulator/1/robot_simulator.py
# Globals for the directions
# Change the values as you see fit
EAST = 1
NORTH = 2
WEST = 3
SOUTH = 4


class Robot:
    def __init__(self, direction=NORTH, x_pos=0, y_pos=0):
        self.direction = direction
        self.coordinates = (x_pos, y_pos)

    def turn_left(self):
        self.direction += 1
        if self.direction == 5:
            self.direction = 1

    def turn_right(self):
        self.direction -= 1
        if self.direction == 0:
            self.direction = 4

    def move(self, instructions):
        for instruction in instructions:
            if instruction == "R":
                self.turn_right()
            elif instruction == "L":
                self.turn_left()
            else:
                x, y = self.coordinates
                if self.direction == NORTH:
                    y += 1
                elif self.direction == EAST:
                    x += 1
                elif self.direction == SOUTH:
                    y -= 1
                elif self.direction == WEST:
                    x -= 1
                self.coordinates = x, y

File: solutions/csharp/robot-simulator/1/RobotSimulator.cs
public enum Direction {
  North,
  East,
  South,
  West
}

public class RobotSimulator {
  private int _x;
  private int _y;
  private Direction _direction;

  public RobotSimulator(Direction direction, int x, int y) {
    _x = x;
    _y = y;
    _direction = direction;
  }

  public Direction Direction {
    get => _direction;
  }

  public int X {
    get => _x;
  }

  public int Y {
    get => _y;
  }

  public void Move(string instructions) {
    foreach (var instruction in instructions) {
      switch (instruction) {
        case 'A':
          Move();
          break;
        case 'L':
          TurnLeft();
          break;
        case 'R':
          TurnRight();
          break;
      }
    }
  }

  private void TurnLeft() {
    var newDirection = (int)_direction - 1;
    if (newDirection < 0) {
      newDirection = (int)Direction.West;
    }

    _direction = (Direction)newDirection;
  }

  private void TurnRight() {
    var newDirection = (int)_direction + 1;
    if (newDirection >= 4) {
      newDirection = (int)Direction.North;
    }

    _direction = (Direction)newDirection;
  }

  private void Move() {
    switch (_direction) {
      case Direction.North:
        _y += 1;
        break;
      case Direction.East:
        _x += 1;
        break;
      case Direction.South:
        _y -= 1;
        break;
      case Direction.West:
        _x -= 1;
        break;
    }
  }
}